torch/nn/quantized/modules/linear.py

```diff
@@ -140,3 +142,39 @@ def _load_from_state_dict(self, state_dict, prefix, local_metadata, strict,
         super()._load_from_state_dict(state_dict, prefix, local_metadata, False,
                                       missing_keys, unexpected_keys, error_msgs)
         return
+
+    @staticmethod
+    def from_float(mod, qparams_dict=None):
+        r"""Create a quantized module from a float module or qparams_dict
+
+            Args: `mod` a float module, either produced by torch.quantization utilities
+            or directly from user
+            Args: `qprams_dict` if `mod` is created directly by user, we will
+            need the dictionary to have entries of quantization parameters for
+            `activation` and `weight`, only torch.per_tensor_affine is supported
+            right now
+        """
+        assert type(mod) == NNLinear, 'nnq.Linear.from_float only works for nn.Linear'
+        if qparams_dict:
+            assert('activation' in qparams_dict)
+            assert('weight' in qparams_dict)
+            act_qparams = qparams_dict['activation']
```

Comment:
why not just pass activation_qparams / wt_qparams directly as args?

also, how will this branch work for non-linear quantization? you just pass qparams, not the quantizer. Should it receive Quantizer in the future? (if quantizer is not ready - I'd rather hide this API, it's always possible to add it later)

Reply:
sure we can remove this branch for now